### PR TITLE
wrap resource name

### PR DESCRIPTION
**Pull request by fanjinfei**

### CODE REVIEW

**Comment by fanjinfei** [comment]
ready for review

File: ckanext/canada/validators.py [hunk 1]
```diff
@@ -110,6 +110,17 @@ def canada_validate_generate_uuid(value):
         raise Invalid(_("Badly formed hexadecimal UUID string"))
 
 
+email_pattern = re.compile(r"^[A-Za-z0-9\.\+_-]+@[A-Za-z0-9\._-]+\.[a-zA-Z]*$")
```
Comment (wardi) [inline ckanext/canada/validators.py]: I suggest https://html.spec.whatwg.org/#e-mail-state-(type=email) (the regexp is below) and add a comment that links to this doc to say where we got it from.

File: ckanext/canada/validators.py [hunk 2]
```diff
@@ -110,6 +110,17 @@ def canada_validate_generate_uuid(value):
...
+def email_validator(value):
+    if value:
+        try:
+            if not email_pattern.match(value):
+                raise Invalid(_('Please enter a valid email address.'))
+        except:
```
Comment (wardi) [inline ckanext/canada/validators.py]: bare `except:`s make @TkTech :anger:

What exception is this catching? TypeError?

File: ckanext/canada/schemas/presets.yaml [hunk 3]
```diff
@@ -3078,6 +3078,8 @@ presets:
     - value: EXE
       mimetype: application/vnd.microsoft.portable-executable
       replaces: [exe]
+    - value: FEFLOW
+      replaces: [feflow]
```
Comment (wardi) [inline ckanext/canada/schemas/presets.yaml]: replaces is just for formats that were in the previous version of the metadata standard. It doesn't apply to any new types

File: ckanext/canada/schemas/presets.yaml [hunk 4]
```diff
@@ -3142,10 +3148,20 @@ presets:
...
     - value: LYR
+    - value: Leapfrog Mining
+      replaces: [leapfrog mining]
+    - value: Leapfrog Hydro
+      replaces: [leapfrog hydro]
+    - value: Leapfrog Viewer
+      replaces: [leapfrog viewer]
     - label: MapInfo
       value: TAB
+    - value: MapInfo Grid
```
Comment (wardi) [inline ckanext/canada/schemas/presets.yaml]: we use the most common extension as the value for these choices (e.g. see TAB above for MapInfo)

Comment (wardi) [inline ckanext/canada/schemas/presets.yaml]: Might need to go back to the client and ask for the list of file extensions along with these format names

**Comment by wardi** [comment]
@fanjinfei please remove the format changes from this PR so that the rest can be merged while we wait.

**Pull request merged**